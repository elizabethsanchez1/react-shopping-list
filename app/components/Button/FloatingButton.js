import PropTypes from 'prop-types';
import React from 'react';
import { Button } from 'react-native-elements';
import styles from '../Form/styles';
import Icon from 'react-native-vector-icons/MaterialIcons';
import theme from '../../styles/theme.style';

const FloatingButton = (props) => {
  return (
    <Button
     //  raised
      // icon={{ name: 'add', type: 'material-icons', buttonStyle: styles.actionButton }}
      title='+'
      backgroundColor={theme.PRIMARY_COLOR}
      borderRadius={50}
      buttonStyle={styles.actionButton}
      fontSize={40}
      onPress={() => props.onClick()}
    />
  )
};

FloatingButton.propTypes = {
  onClick: PropTypes.func,
};

export default FloatingButton;
