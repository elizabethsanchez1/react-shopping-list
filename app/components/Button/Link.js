import React from 'react';
import { Text, StyleSheet } from 'react-native';
import PropTypes from 'prop-types';
import theme from '../../styles/theme.style';

const styles = StyleSheet.create( {
  text: {
    fontSize: theme.FONT_SIZE_MEDIUM,
    color: theme.ACTIVE_TAB_COLOR,
  },
} );

const Link = ( { title, onPress } ) => {
  return (
    <Text style={ styles.text } onPress={ onPress }>
      { title }
    </Text>
  );
};

Link.propTypes = {
  title: PropTypes.string,
  onPress: PropTypes.func,
};

export default Link;
