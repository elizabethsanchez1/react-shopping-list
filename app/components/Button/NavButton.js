import React, { Component } from 'react';
import { Button } from 'react-native-elements';
import { StyleSheet } from 'react-native';
import PropTypes from 'prop-types';
import theme from '../../styles/theme.style';

const styles = StyleSheet.create( {
  navigationButton: { backgroundColor: 'transparent' },
  text: {
    fontSize: 18,
  },
} );

class NavButton extends Component {
  constructor( props ) {
    super( props );
    this.state = {
      isRunning: false,
    };
  }

  handleClick = () => {
    if ( !this.props.isDisabled ) {

      if ( !this.state.isRunning ) {
        this.props.onPress();
      }

      this.setState( { isRunning: true } );
      setTimeout( () => {
        this.setState( { isRunning: false } );
      }, 10000 );
    }
  };

  render() {
    const { isDisabled, title } = this.props;

    return (
      <Button
        buttonStyle={ styles.navigationButton }
        color={ isDisabled ? theme.DISABLED_TEXT_COLOR : theme.ACTIVE_TAB_COLOR }
        textStyle={ styles.text }
        title={ title }
        onPress={ () => this.handleClick() }
      />
    );
  }
}

NavButton.propTypes = {
  title: PropTypes.string,
  isDisabled: PropTypes.bool,
  onPress: PropTypes.func,
};

export default NavButton;
