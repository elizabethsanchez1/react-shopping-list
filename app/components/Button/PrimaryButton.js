import PropTypes from 'prop-types';
import React from 'react';
import { Button } from 'react-native-elements';
import styles from '../Form/styles';
import theme from '../../styles/theme.style';

const PrimaryButton = (props) => {
  if (props.loading) {
    return (
      <Button
        large
        {...props}
        buttonStyle={[styles.primaryButton, props.buttonStyle]}
        fontFamily={
          (props.fontFamily) ? props.fontFamily : theme.PRIMARY_FONT_FAMILY
        }
        fontWeight={
          (props.fontWeight) ? props.fontWeight : theme.FONT_WEIGHT_MEDIUM
        }
        color={
          (props.disabled) ? theme.DISABLED_TEXT_COLOR  : theme.PRIMARY_FONT_COLOR
        }
        disabled={props.loading}
        title=""
      />
    );
  }


  return (
    <Button
      large
      buttonStyle={[styles.primaryButton, props.buttonStyle]}
      fontFamily={
        (props.fontFamily) ? props.fontFamily : theme.PRIMARY_FONT_FAMILY
      }
      fontSize={
        (props.fontSize) ? props.fontSize : theme.FONT_SIZE_LARGE
      }
      fontWeight={
        (props.fontWeight) ? props.fontWeight : theme.FONT_WEIGHT_MEDIUM
      }
      color={
        (props.disabled) ? theme.DISABLED_TEXT_COLOR  : theme.PRIMARY_FONT_COLOR
      }
      disabledStyle={styles.disabledStyle}
      disabledTextStyle={{color: 'blue'}}
      title={props.title}
      containerViewStyle={props.containerViewStyle}
      onPress={props.onPress}
      // fontWeight={[styles.fontWeight, props.fontWeight]}
    />
  );
};

PrimaryButton.propTypes = {
  buttonStyle: PropTypes.object,
  containerViewStyle: PropTypes.object,
  title: PropTypes.string,
  fontFamily: PropTypes.string,
  fontWeight: PropTypes.string,
  fontSize: PropTypes.string,
  loading: PropTypes.bool,
  disabled: PropTypes.bool,
  onPress: PropTypes.func,
};

export default PrimaryButton;
