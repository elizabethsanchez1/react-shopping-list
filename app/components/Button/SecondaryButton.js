import PropTypes from 'prop-types';
import React from 'react';
import { TouchableOpacity, Text } from 'react-native';
import styles from '../Form/styles';

const SecondaryButton = (props) => {
  return (
    <TouchableOpacity
      onPress={props.onPress}
      style={[styles.secondaryButton, props.style]}
    >
      <Text style={styles.secondaryButtonText}>
        {props.title}
      </Text>
    </TouchableOpacity>
  );
};

SecondaryButton.propTypes = {
  title: PropTypes.string,
  style: PropTypes.object,
  onPress: PropTypes.func,
};

export default SecondaryButton;
