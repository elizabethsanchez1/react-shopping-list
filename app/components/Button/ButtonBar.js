import React from 'react';
import PropTypes from 'prop-types';
import { View, StyleSheet } from 'react-native';
import PrimaryButton from './PrimaryButton';

const styles = StyleSheet.create( {
  container: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  button: {
    padding: 15,
  },
  buttonContainer: {
    marginLeft: 0,
    marginRight: 0,
  }
} );

const ButtonBar = ( { title1, onPress1, title2, onPress2 } ) => (
  <View style={ styles.container }>
    <PrimaryButton
      title={ title1 }
      onPress={ onPress1 }
      buttonStyle={ styles.button }
      containerViewStyle={ styles.buttonContainer }
    />

    <PrimaryButton
      title={ title2 }
      onPress={ onPress2 }
      buttonStyle={ styles.button }
      containerViewStyle={ styles.buttonContainer }
    />
  </View>
);

ButtonBar.propTypes = {
  title1: PropTypes.string,
  title2: PropTypes.string,
  onPress1: PropTypes.func,
  onPress2: PropTypes.func,
};

export default ButtonBar;
