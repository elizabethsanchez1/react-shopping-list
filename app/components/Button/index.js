import PrimaryButton from './PrimaryButton';
import SecondaryButton from './SecondaryButton';
import FloatingButton from './FloatingButton';
import Link from './Link';
import ButtonBar from './ButtonBar';


export { PrimaryButton, SecondaryButton, FloatingButton, Link, ButtonBar };
