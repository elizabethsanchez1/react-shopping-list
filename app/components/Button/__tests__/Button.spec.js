import 'react-native';
import React from 'react';
import { Text } from 'react-native';
import renderer from 'react-test-renderer';
import { FloatingButton, Link, PrimaryButton, SecondaryButton } from "../../Button";

describe( 'Button Components', () => {

  it( 'Primary Button visually looks the same', () => {
    const tree = renderer.create(
      <PrimaryButton
        title="PRIMARY"
      />
    ).toJSON();

    expect( tree ).toMatchSnapshot();
  } );

  it( 'Primary Button with loading visually looks the same', () => {
    const tree = renderer.create(
      <PrimaryButton
        title="PRIMARY"
        loading={ true }
      />
    ).toJSON();

    expect( tree ).toMatchSnapshot();
  } );

  it( 'Primary Button disabled visually looks the same', () => {
    const tree = renderer.create(
      <PrimaryButton
        title="PRIMARY"
        disabled
      />
    ).toJSON();

    expect( tree ).toMatchSnapshot();
  } );


  it( 'Secondary Button visually looks the same', () => {
    const tree = renderer.create(
      <SecondaryButton
        title="SECONDARY"
      />
    ).toJSON();

    expect( tree ).toMatchSnapshot();
  } );


  it( 'Floating Button visually looks the same', () => {
    const tree = renderer.create(
      <FloatingButton/>
    ).toJSON();

    expect( tree ).toMatchSnapshot();
  } );


  it( 'Link Button visually looks the same', () => {
    const tree = renderer.create(
      <Link/>
    ).toJSON();

    expect( tree ).toMatchSnapshot();
  } );
} );




