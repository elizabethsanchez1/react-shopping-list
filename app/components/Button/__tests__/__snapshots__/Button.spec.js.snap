// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Components Floating Button visually looks the same 1`] = `
<View
  style={
    Array [
      Object {
        "marginLeft": 15,
        "marginRight": 15,
      },
      undefined,
      undefined,
      Object {
        "borderRadius": 50,
      },
    ]
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={null}
  >
    <View
      pointerEvents="box-only"
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#9E9E9E",
            "flexDirection": "row",
            "justifyContent": "center",
            "padding": 19,
          },
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Object {
            "backgroundColor": "#364590",
          },
          Object {
            "borderRadius": 50,
          },
          Object {
            "padding": 12,
          },
          undefined,
          undefined,
          undefined,
          Object {
            "borderRadius": 50,
            "elevation": 1,
            "height": 60,
            "padding": 5,
            "paddingBottom": 10,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.5,
            "shadowRadius": 2,
            "width": 60,
          },
          undefined,
          undefined,
        ]
      }
    >
      <Text
        style={
          Array [
            Object {},
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Array [
              Object {
                "color": "white",
                "fontSize": 20,
              },
              undefined,
              Object {
                "fontSize": 17.5,
              },
              Object {
                "fontSize": 40,
              },
              undefined,
              undefined,
              undefined,
            ],
          ]
        }
      >
        +
      </Text>
    </View>
  </View>
</View>
`;

exports[`Button Components Link Button visually looks the same 1`] = `
<Text
  style={
    Object {
      "color": "#72DEFF",
      "fontSize": 16,
    }
  }
/>
`;

exports[`Button Components Primary Button disabled visually looks the same 1`] = `
<View
  style={
    Array [
      Object {
        "marginLeft": 15,
        "marginRight": 15,
      },
      undefined,
      undefined,
      undefined,
    ]
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={null}
  >
    <View
      pointerEvents="box-only"
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#9E9E9E",
            "flexDirection": "row",
            "justifyContent": "center",
            "padding": 19,
          },
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          false,
          undefined,
          undefined,
          undefined,
          Array [
            Object {
              "backgroundColor": "#364590",
              "borderRadius": 4,
              "elevation": 1,
              "marginTop": 20,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 3,
                "width": 0,
              },
              "shadowOpacity": 0.8,
              "shadowRadius": 2,
            },
            undefined,
          ],
          undefined,
          undefined,
        ]
      }
    >
      <Text
        style={
          Array [
            Object {},
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Array [
              Object {
                "color": "white",
                "fontSize": 20,
              },
              Object {
                "color": "#a7a8a8",
              },
              false,
              Object {
                "fontSize": 18,
              },
              undefined,
              Object {
                "fontWeight": "600",
              },
              Object {
                "fontFamily": "RobotoCondensed-Regular",
              },
            ],
          ]
        }
      >
        PRIMARY
      </Text>
    </View>
  </View>
</View>
`;

exports[`Button Components Primary Button visually looks the same 1`] = `
<View
  style={
    Array [
      Object {
        "marginLeft": 15,
        "marginRight": 15,
      },
      undefined,
      undefined,
      undefined,
    ]
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={null}
  >
    <View
      pointerEvents="box-only"
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#9E9E9E",
            "flexDirection": "row",
            "justifyContent": "center",
            "padding": 19,
          },
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          false,
          undefined,
          undefined,
          undefined,
          Array [
            Object {
              "backgroundColor": "#364590",
              "borderRadius": 4,
              "elevation": 1,
              "marginTop": 20,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 3,
                "width": 0,
              },
              "shadowOpacity": 0.8,
              "shadowRadius": 2,
            },
            undefined,
          ],
          undefined,
          undefined,
        ]
      }
    >
      <Text
        style={
          Array [
            Object {},
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Array [
              Object {
                "color": "white",
                "fontSize": 20,
              },
              Object {
                "color": "#FFF",
              },
              false,
              Object {
                "fontSize": 18,
              },
              undefined,
              Object {
                "fontWeight": "600",
              },
              Object {
                "fontFamily": "RobotoCondensed-Regular",
              },
            ],
          ]
        }
      >
        PRIMARY
      </Text>
    </View>
  </View>
</View>
`;

exports[`Button Components Primary Button with loading visually looks the same 1`] = `
<View
  style={
    Array [
      Object {
        "marginLeft": 15,
        "marginRight": 15,
      },
      undefined,
      undefined,
      undefined,
    ]
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={null}
  >
    <View
      pointerEvents="box-only"
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#9E9E9E",
            "flexDirection": "row",
            "justifyContent": "center",
            "padding": 19,
          },
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          false,
          undefined,
          undefined,
          undefined,
          Array [
            Object {
              "backgroundColor": "#364590",
              "borderRadius": 4,
              "elevation": 1,
              "marginTop": 20,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 3,
                "width": 0,
              },
              "shadowOpacity": 0.8,
              "shadowRadius": 2,
            },
            undefined,
          ],
          Object {
            "backgroundColor": "#dadee0",
          },
          undefined,
        ]
      }
    >
      <ActivityIndicator
        animating={true}
        color="#FFF"
        hidesWhenStopped={true}
        size="large"
        style={
          Array [
            Object {
              "height": 0,
              "marginHorizontal": 10,
            },
            undefined,
          ]
        }
      />
      <Text
        style={
          Array [
            Object {},
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Array [
              Object {
                "color": "white",
                "fontSize": 20,
              },
              Object {
                "color": "#FFF",
              },
              false,
              undefined,
              undefined,
              Object {
                "fontWeight": "600",
              },
              Object {
                "fontFamily": "RobotoCondensed-Regular",
              },
            ],
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</View>
`;

exports[`Button Components Secondary Button visually looks the same 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderColor": "#fff",
      "borderRadius": 5,
      "borderWidth": 1,
      "opacity": 1,
      "padding": 12,
    }
  }
>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "color": "#FFF",
      }
    }
  >
    SECONDARY
  </Text>
</View>
`;
